pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -544,24 +550,25 @@ public final synchronized void readEntriesComplete(List<Entry> entries, Object c
         if (serviceConfig.isDispatcherDispatchMessagesInSubscriptionThread()) {
             // setting sendInProgress here, because sendMessagesToConsumers will be executed
             // in a separate thread, and we want to prevent more reads
```

Comment:
this comment is now outdated since `sendInProgress = true;` was removed. Was it intentional to remove `sendInProgress = true`?

Reply:
It's better to move it to line 565.

Reply:
It does change the behavior that is explained in the comment "setting sendInProgress here, because sendMessagesToConsumers will be executed in a separate thread, and we want to prevent more reads." 
You didn't remove the comment.

Reply:
Yes, you are right. I'll try to revert this line, but I still have one concern.
Setting the `true` here is risky, and if there is a problem with the executor, it will cause the value to never become `false`.

I'm not sure if we have to separate it to different PRs, one is reverting this line, and another is to fix this line-involved problem.

Reply:
> Setting the true here is risky, and if there is a problem with the executor, it will cause the value to never become false.

that's a valid concern (but perhaps not something that would be a problem in practice). 
The more severe issue is that logic won't work as intended by @eolivelli in #16812 if you remove the setting to true.

Reply:
Yes, I will revert it to its original behaviour,

Reply:
Did this get resolved in another PR?

Reply:
I opened https://github.com/apache/pulsar/pull/17029 to fix this. It appears to have introduced a regression that led to several flaky tests. Please review the PR.

Reply:
Hi, @michaeljmarshall  The revert PR already exist. #17018

Reply:
Thanks for confirming!